internal/dnsforward/dnsforward.go

```diff
@@ -480,7 +481,21 @@ func (s *Server) Prepare(config *ServerConfig) error {
 
 	// Initialize ipset configuration
 	// --
-	err := s.ipset.init(s.conf.IpsetList)
+	var ipsets []string
+	if s.conf.IpsetListFileName != "" {
+		data, err := os.ReadFile(s.conf.IpsetListFileName)
+		if err != nil {
+			return err
+		}
+
+		ipsets = stringutil.SplitTrimmed(string(data), "\n")
+
+		log.Debug("dns: using %d ipset list from file %s", len(ipsets), s.conf.IpsetListFileName)
+	} else {
+		ipsets = s.conf.IpsetList
+	}
+
+	err := s.ipset.init(ipsets)
 	if err != nil {
 		return err
 	}
```

Comment:
This increases the [cyclomatic complexity][1] of method `Prepare` too much.  On `master`, we have finally reduced the complexity of all methods in package `dnsforward` to 10 and below.  Please put what is now on lines 484 to 501 to a separate method.  Do not forget about error context.

[1]: https://en.wikipedia.org/wiki/Cyclomatic_complexity